nvflare/app_opt/xgboost/histogram_based_v2/sec/client_handler.py

```diff
@@ -291,11 +345,28 @@ def _process_after_all_gather_v(self, fl_ctx: FLContext):
         result = self.data_converter.encode_aggregation_result(final_result, fl_ctx)
         fl_ctx.set_prop(key=Constant.PARAM_KEY_RCV_BUF, value=result, private=True, sticky=False)
 
+    def _process_after_all_gather_v_horizontal(self, fl_ctx: FLContext):
+        encrypted_histograms = fl_ctx.get_prop(Constant.PARAM_KEY_RCV_BUF)
+        rank = fl_ctx.get_prop(Constant.PARAM_KEY_RANK)
+        if not isinstance(encrypted_histograms, CKKSVector):
+            return self._abort(f"rank {rank}: expect a CKKSVector but got {type(encrypted_histograms)}", fl_ctx)
+
+        histograms = encrypted_histograms.decrypt(secret_key=self.tenseal_context.secret_key())
+        result = self.data_converter.encode_histograms_result(histograms, fl_ctx)
+        fl_ctx.set_prop(key=Constant.PARAM_KEY_RCV_BUF, value=result, private=True, sticky=False)
+
     def handle_event(self, event_type: str, fl_ctx: FLContext):
         if event_type == EventType.START_RUN:
             self.public_key, self.private_key = generate_keys(self.key_length)
             self.encryptor = Encryptor(self.public_key, self.num_workers)
             self.decrypter = Decrypter(self.private_key, self.num_workers)
             self.adder = Adder(self.num_workers)
+            try:
```

Comment:
Not sure this is a good use case if the application is not horizontal secure xgb - there will always be a msg like: 
Can't load tenseal context, horizontal secure XGBoost is not supported: {ex}

Maybe we could check self.tenseal_context_file. If it's specified, then we treat it as horizontal secure; otherwise we don't load it.

Reply:
Made tenseal optional. If tenseal not available, there will be no errors. Hortizontal secure XGBoost is not supported in this case.

The vertical works on Mac now with this change.